code/game/objects/items/dualsaber.dm

```diff
@@ -69,9 +74,30 @@
 /obj/item/dualsaber/proc/on_unwield(obj/item/source, mob/living/carbon/user)
 	update_weight_class(initial(w_class))
 	hitsound = SFX_SWING_HIT
+	heat = initial(heat)
 	STOP_PROCESSING(SSobj, src)
 	set_light_on(FALSE)
 
+/obj/item/dualsaber/tool_use_check(mob/living/user, amount, heat_required)
+	if(!HAS_TRAIT(src, TRAIT_WIELDED))
+		balloon_alert(user, "not active!")
+		return FALSE
+	if(heat < heat_required)
+		balloon_alert(user, "not hot enough!")
+		return FALSE
+	return TRUE
+
+/obj/item/dualsaber/use(used)
+	return HAS_TRAIT(src, TRAIT_WIELDED)
```

Comment:
use() is supposed to consume material no? what's going on with its use in this pr?

Reply:
use means its being used, stuff like welders works the same except it consumes fuel. i dont see the problem with this if we're not consuming anything